### ISSUE TITLE
Issue: OBS doesnt understand values

### ISSUE THREAD

**Issue by triceratops1** [openSUSE/obs-build#350]
if a  program has expressions as dependencies
example
depends ('perl>5.15') 

obs cant read

**Comment by dg0yt** [comment]
According to the syntax, is this issue for Arch Linux? With the Arch Community repository I'm facing the issue that I can't use libgeotiff because OBS claims that "nothing provides libjpeg>=7 needed by libgeotiff". libjpeg=8.1.2 is [provided by libjpeg-turbo](https://www.archlinux.org/packages/extra/i686/libjpeg-turbo/) which should suffice. But even a dummy libjpeg package doesn't help to solve this particular dependency for OBS.

**Comment by dg0yt** [comment]
@nyancat18 ping...

**Comment by dg0yt** [comment]
Oh, now I even find unchanged PKGBUILDs breaking which used to work in the past...

**Comment by mlschroe** [comment]
Where?

**Comment by dg0yt** [comment]
Test case: https://build.opensuse.org/package/show/home:dg0yt:branches:home:dg0yt:staging/openorienteering-mapper

**Comment by mlschroe** [comment]
Thanks! That was an unforseen side effect of merging pull request 5 in perl-BSSolv (https://github.com/openSUSE/perl-BSSolv/pull/5). I knew that could cause trouble so I added an extra check so the change was only done if the binary type was rpm. Unfortunately the OBS code didn't set the correct type for arch linux, so the change intended for just rpm did break arch linux building.

It's now fixed and your package builds again.

**Comment by dg0yt** [comment]
Solved for me, thanks.

**Comment by dg0yt** [comment]
Not sure if it is related, but I found more PKGBUILDS which are unresolvable due to a versioned dependency within Arch Core, gcc-fortran 6.3.1-2 dependency on ```gcc=6.3.1-2```:
In OBS: https://build.opensuse.org/project/monitor/home:Minerva_W:Science?arch_x86_64=1&defaults=0&repo_Arch_Extra=1&unresolvable=1
gcc in Arch core: https://www.archlinux.org/packages/?sort=&repo=Core&q=gcc

**Comment by mlschroe** [comment]
Yes, it's related. The code that parses the repository data for "download on demand" repos suffers from the same change. I'll fix it asap.

**Comment by mlschroe** [comment]
Now fixed in the repository parser (obs-build, commit 996f6087188cf0ee6e04dcd98995ec5eaa2d4e8d).
Thanks for reporting this.
